lib/GLPI/Agent/SNMP/MibSupport/LinuxAppliance.pm

```diff
@@ -257,6 +268,52 @@ sub run {
 
     my $firmware;
     if ($manufacturer eq 'Synology') {
+        
+        my $diskModels = $self->walk(dsmDiskModel)
+        or return;
+
+        my $diskNames = $self->walk(dsmDiskName)
+        or return;
+
+        my $volumesNames = $self->walk(dsmRaidName)
+        or return;
+
+        my $volumesFreeSizes = $self->walk(dsmRaidFreeSize)
+        or return;
+
+        my $volumesTotalSizes = $self->walk(dsmRaidTotalSize)
+        or return;
+
+        foreach my $key (keys(%{$diskModels})){
+
+            my $storage = {
+            TYPE    => 'disk',
+            };
+
+            my $name = trimWhitespace(getCanonicalString($diskNames->{$key}));
+            my $model = $diskModels->{$key};
+            my $manufacturer = getCanonicalManufacturer($model);
+            $storage->{MANUFACTURER} = $manufacturer; 
+            $storage->{MODEL} = $model . " - " . $name;
+            push @{$device->{STORAGES}}, $storage
+            if $storage->{MODEL};
+        }
+
+
+        foreach my $key (keys(%{$volumesNames}))
+        {
+            my $name = trimWhitespace(getCanonicalString($volumesNames->{$key}));
+
+            my $volumes = {
+            VOLUMN => $name,
+            FREE => getCanonicalSize("$volumesFreeSizes->{$key} bytes"),
+            TOTAL => getCanonicalSize("$volumesTotalSizes->{$key} bytes")
+            };
+
+            push @{$device->{DRIVES}}, $volumes
+            if $volumes->{VOLUMN};
```

Comment:
Same issue than previous, just skip on no defined name, and again, you have wrong indentation on the structure definition:
```suggestion
            my $name = trimWhitespace(getCanonicalString($volumesNames->{$key}))
                or next;

            my $volumes = {
                VOLUMN => $name,
                FREE => getCanonicalSize("$volumesFreeSizes->{$key} bytes"),
                TOTAL => getCanonicalSize("$volumesTotalSizes->{$key} bytes")
            };

            push @{$device->{DRIVES}}, $volumes;
```
But, you miss another point, you have no guarantee `$volumesFreeSizes->{$key}` & `$volumesTotalSizes->{$key}` are defined. So use better the following code:
```suggestion
            my $name = trimWhitespace(getCanonicalString($volumesNames->{$key}))
                or next;

            my $volumn = {
                VOLUMN => $name,
            };
            $volumn->{FREE} = getCanonicalSize("$volumesFreeSizes->{$key} bytes")
                if $volumesFreeSizes->{$key};
            $volumn->{TOTAL} => getCanonicalSize("$volumesTotalSizes->{$key} bytes")
                if $volumesTotalSizes->{$key};

            push @{$device->{DRIVES}}, $volumn;
```